spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -82,6 +82,8 @@ public Long endSnapshotId() {
     return confParser.longConf().option(SparkReadOptions.END_SNAPSHOT_ID).parseOptional();
   }
 
+  /** @deprecated will be removed in 1.3.0; specify read branch in the table identifier */
```

Comment:
Can we at-least support reading by spark options. This we because we already support time travel for the same and branch/tag is also a part of time travel.

Reply:
There's probably a way to reconcile having both read options and querying via `select * from table.branch_identifier`. Right now it's handled by just validating that if a read option is set, validate the identifier branch is the same.

That being said I don't think it's worth it in the long run for the user experience, because then for Dataframes there's multiple ways for a user to query a branch (in the identifier and for read options) so it gets a bit confusing there for users. We then have the maintainability issues for handling cases where read options and identifiers are set. Considering we now know that having the branch in the identifier is the way to go for the write case, it makes sense to me to have the same single approach for the read case.